health/service/src/main/java/com/cerner/beadledom/health/HttpHealthDependency.java

```diff
@@ -0,0 +1,54 @@
+package com.cerner.beadledom.health;
+
+import java.io.IOException;
+import java.util.Objects;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Implementation of {@link HealthDependency} to check the health of any HTTP service.
+ *
+ * @author Venkatesh Sridharan.
+ */
+public class HttpHealthDependency extends HealthDependency {
+  private final String availabilityUrl;
+  private final String name;
+  private final OkHttpClient client;
+
+  /**
+   * Constructs an instance of {@link HttpHealthDependency} with the given parameters.
+   *
+   * @param availabilityUrl String representing URL of the availability health check.
+   * @param name String representing name of the dependency
+   * @param isPrimary Boolean representing whether the dependency is primary or not.
+   */
+  public HttpHealthDependency(String availabilityUrl, String name, boolean isPrimary) {
+    super(isPrimary);
+    Objects.requireNonNull(availabilityUrl);
+    Objects.requireNonNull(name);
+
+    this.availabilityUrl = availabilityUrl;
+    this.name = name;
+    this.client = new OkHttpClient.Builder().build();
+  }
+
+  @Override
+  public HealthStatus checkAvailability() {
+    Request request = new Request.Builder().url(availabilityUrl).head().build();
+
+    try (Response response = client.newCall(request).execute()) {
+      if (response.isSuccessful()) {
+        return HealthStatus.create(200, getName() + " is available.");
+      }
+      return HealthStatus.create(503, getName() + " is not available.");
```

Comment:
probably use the real status from the call and not 503.

Reply:
Will the consuming class for this change its logic in anyway if this is a 4xx instead of a 503?

Reply:
If they choose to. Internally, on the server side, we check if it is a 2XX status [1]. Either way, it is more appropriate to return the actual status, 

```java
  /**
   * Returns the HTTP status code returned by the dependency, as seen from the location of the
   * health check resource (eg. `200`).
   *
   * <p>For dependencies without an HTTP health check, this should simulate what such a health check
   * would return - e.g. 200 for healthy, 503 for unavailable.
   */
  public abstract int getStatus();
```

[1] https://github.com/cerner/beadledom/blob/master/health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java#L155

Reply:
The RFC for the dependencies endpoint says this too "SHOULD simply relay HTTP status of dependencies basic availability check"

Reply:
1105d152, I am still returning 503 when an IOException is thrown. Is that okay?